src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/ContainerBuilderExtensions.cs

```diff
@@ -0,0 +1,190 @@
+        }
+
+        public static void RegisterEndpoints(this ContainerBuilder builder)
+        {
+            builder.RegisterEndpoint<RegistrationEndpoint>();
+            builder.RegisterEndpoint<FlatContainerEndpoint>();
+            builder.RegisterEndpoint<CatalogEndpoint>();
+        }
+
+        private static void RegisterEndpoint<T>(this ContainerBuilder builder)
+            where T : IEndpoint
+        {
+            builder
+                .RegisterType<T>()
+                .AsSelf()
+                .As<IEndpoint>();
+
+            builder
+                .RegisterType<EndpointValidator<T>>()
+                .AsSelf()
+                .As<IAggregateValidator>();
+        }
+
+        public static void RegisterValidators(this ContainerBuilder builder)
+        {
+            // Catalog validators
+            builder.RegisterValidator<CatalogEndpoint, PackageHasSignatureValidator>();
+
+            // Registration validators
+            builder.RegisterValidator<RegistrationEndpoint, RegistrationExistsValidator>();
+            builder.RegisterValidator<RegistrationEndpoint, RegistrationIdValidator>();
+            builder.RegisterValidator<RegistrationEndpoint, RegistrationListedValidator>();
+            builder.RegisterValidator<RegistrationEndpoint, RegistrationRequireLicenseAcceptanceValidator>();
+            builder.RegisterValidator<RegistrationEndpoint, RegistrationVersionValidator>();
+
+            // Flat-container validators
+            builder.RegisterValidator<FlatContainerEndpoint, PackageIsRepositorySignedValidator>();
+        }
+
+        private static void RegisterValidator<TEndpoint, TValidator>(this ContainerBuilder builder)
+            where TEndpoint : IEndpoint
+            where TValidator : IValidator<TEndpoint>
+        {
+            builder
+                .RegisterType<TValidator>()
+                .As<IValidator<TEndpoint>>();
+        }
+
+        public static void RegisterSourceRepositories(
+            this ContainerBuilder builder,
+            string galleryUrl,
+            string indexUrl)
+        {
+            if (string.IsNullOrEmpty(galleryUrl))
+            {
+                throw new ArgumentException(Strings.ArgumentMustNotBeNullOrEmpty, nameof(galleryUrl));
+            }
+
+            if (string.IsNullOrEmpty(indexUrl))
+            {
+                throw new ArgumentException(Strings.ArgumentMustNotBeNullOrEmpty, nameof(indexUrl));
+            }
+
+            builder
+                .RegisterInstance(new PackageSource(galleryUrl))
+                .Keyed<PackageSource>(FeedType.HttpV2);
+
+            builder
+                .RegisterInstance(new PackageSource(indexUrl))
+                .Keyed<PackageSource>(FeedType.HttpV3);
+
+            builder.RegisterV2ResourceProviders();
+            builder.RegisterV3ResourceProviders();
+
+            foreach (var type in new[] { FeedType.HttpV2, FeedType.HttpV3 })
+            {
```

Comment:
This `foreach` seems a little overkill. Let's just call `RegisterSourceRepository` twice?